src/lib/parse/CompoundValueNode.cpp

```diff
@@ -270,7 +211,7 @@ std::string CompoundValueNode::finish(void)
     show("");
   }
 
-  check();  // sets 'error' for toplevel node
+  error = check("");  // sets 'error' for toplevel node
 
   return error;
```

Comment:
Unless the variable 'error' is used elsewhere, perhaps we can remove it completely and simply `return check("");` ?
[ Even better would be to make the function return bool ... but that's probably a lot of work ]